src/mon/MonClient.cc

```diff
@@ -593,28 +704,125 @@ string MonClient::_pick_random_mon()
   }
 }
 
+//ldout(cct, 10) << "[_multiple_reopen_session] : " << dendl;
+void MonClient::_multiple_reopen_session(int rank, string name)
+{
+  assert(monc_lock.is_locked());
+  ldout(cct, 10) << "[_multiple_reopen_session] : on rank [" << rank << "] name [" << name << "]" << dendl;
+
+  // get the initial mon if one is selected based on inputs
+  string mon;
+  if (rank < 0 && name.length() == 0) {
+    //mon = _pick_random_mon();
+    mon = monmap.get_name(0);
+  } else if (name.length()) {
+    mon = name;
+  } else {
+    mon = monmap.get_name(rank);
+  }
+
+  int attempt = monmap.size(); //FIXREF
+
+  //mark down all connections, reset the state_map
+  _mark_down_all();
+
+  // create and fill the connection vector
+  vector<ConnectionRef> con_vec;
```

Comment:
if the a certain mon is specified, do we still want to initiate multiple connections to all mons?